dali/kernels/imgproc/resample/resampling_setup.cc

```diff
@@ -98,80 +106,224 @@ SeparableResamplingSetup<spatial_ndim>::ComputeScaleAndROI(
   return roi;
 }
 
+/**
+ * @brief Calculates optimum processing order based on input/output sizes and filter support.
+ *
+ * The sizes of intermediate storage and time taken to compute the intermediate images
+ * may depend on the order - i.e. if downscaling only one axis, it's beneficial to resample that
+ * axis first, so that intermediate image is smaller.
+ */
+template <int ndim>
+struct ProcessingOrderCalculator {
+  ProcessingOrderCalculator(ivec<ndim> in_size, ivec<ndim> out_size, ivec<ndim> filter_support)
+  : in_size(in_size), out_size(out_size), filter_support(filter_support) {
+  }
+
+  ProcessingOrder<ndim> operator()() {
+    for (int i = 0; i < ndim; i++)
+      best_order[i] = i;
+    min_cost = 1e+30f;
+    axis_mask = {};
+    curr_size = in_size;
+    run(0);
+    return best_order;
+  }
+
+ private:
+  float compute_cost_mul(int axis) {
+    // y-axis is likely to be the cheapest
+    return axis == 0 ? 1.4f : axis > 1 ? 1.2f : 1.0f;
+  }
+
+  static constexpr float size_bias = 3;
```

Comment:
What the size_bias is?

Reply:
It's how much smaller size is preferred over smaller compute. Usually this doesn't matter, since typically smaller size corresponds to smaller or equal compute.